cadence/src/builder.rs

```diff
@@ -382,6 +461,73 @@ where
         self
     }
 
+    /// Add a container_id to this metric.
+    pub fn with_container_id(mut self, container_id: &'m str) -> Self {
+        if let BuilderRepr::Success(ref mut formatter, _) = self.repr {
+            formatter.with_container_id(container_id);
+        }
+        self
+    }
+
+    pub(crate) fn with_container_id_opt(mut self, container_id: Option<&'m str>) -> Self {
+        if let BuilderRepr::Success(ref mut formatter, _) = self.repr {
+            if let Some(container_id) = container_id {
+                formatter.with_container_id(container_id);
+            }
+        }
+        self
+    }
+
+    /// Add a timestamp to this metric.
+    /// # Example
+    ///
+    /// ```
+    /// use cadence::prelude::*;
+    /// use cadence::{StatsdClient, NopMetricSink, Metric};
+    /// use std::time::{SystemTime, UNIX_EPOCH};
+    ///
+    /// let client = StatsdClient::from_sink("some.prefix", NopMetricSink);
+    /// let timestamp = SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_secs();
+    /// let res = client.count_with_tags("some.key", 1)
+    ///   .with_timestamp(timestamp)
+    ///   .try_send();
+    ///
+    /// assert_eq!(
+    ///   "some.prefix.some.key:1|c|T".to_string() + &timestamp.to_string(),
+    ///  res.unwrap().as_metric_str()
+    /// );
+    /// ```
+    pub fn with_timestamp(mut self, timestamp: u64) -> Self {
+        if let BuilderRepr::Success(ref mut formatter, _) = self.repr {
+            formatter.with_timestamp(timestamp);
+        }
+
+        self
+    }
+
+    /// Add a sampling rate to this metric.
+    /// # Example
```

Comment:
Can you mention that this must be between 0 and 1?

Reply:
done